### DISCUSSION TITLE
Color of the robot: Red or Blue?

### DISCUSSION THREAD

**Discussion by MJavadZallaghi** [cyberbotics/wrestling#38] [Q&A] [answered]
Hi dear @omichel.
I noticed that the some times my robot is blue, and some times it's red. for example, at the current moment, my robot is blue against my top and my below competitors. I want to know how the team color is determined in every game. Is it random? Or is it based on the controller update date? In low-level image processing, the color can be important in some algorithms.
Thanks.

**Comment by omichel** [comment, marked as answer]
Sure, the color of the robot is important for image processing in this game. Your robot will be red if it is attacking, e.g., if you started a series of games by pushing a commit on GitHub. It will be blue if it is defending, e.g., if someone else started a series of games by pushing a commit on their GitHub repository and it turns out that their robot is playing a game against yours.

During the finals to be held from May 30th to June 2nd, the colors will be chosen randomly.

It is indeed possible for you to know the color of your robot with this sample code:

```python
    if self.getName() == 'participant'
        print('I am the red one')
    else:  # the name of the robot should be 'opponent'
        print('I am the blue one')
```

**Reply by MJavadZallaghi** [reply]
Thanks, In the final stage of the competitions. the controller which will be assigned as participant or opponent, will be selected randomly? 
I want to use `self.getName()`, and I want to be sure that for any scenario of random color assignment, this code can detect my color based on the controller name.

**Reply by omichel** [reply]
> Thanks, In the final stage of the competitions. the controller which will be assigned as participant or opponent, will be selected randomly?

Yes, exactly.

> I want to use `self.getName()`, and I want to be sure that for any scenario of random color assignment, this code can detect my color based on the controller name.

Yes, `self.getName()` will provide you the information needed to determine your color in any case, including the scenario of random color assignment.
